pills/17-nixpkgs-overriding-packages.xml

```diff
@@ -44,31 +44,20 @@
   <section>
     <title>Fixed point</title>
     <para>
-      The fixed point with lazy evaluation is crippling but about necessary in a language like Nix. It lets us achieve something similar to what we'd do imperatively.
+      A fixed point is a common pattern in lazy languages to ensure that functions that would otherwise return non fully evaluated values do return fully evaluated values. Let's unpack this idea. Remember that in Nix functions are lazy. Consider:
```

Comment:
> A fixed point is a common pattern in lazy languages to ensure that functions that would otherwise return non fully evaluated values do return fully evaluated values.

I am not sure that forcing strict evaluation (i.e. [`builtins.seq`](https://nixos.org/manual/nix/stable/language/builtins#builtins-seq)) has anything to do with fixed point.

But even if it had, I believe the main purpose of fix point operators in functional languages is (mutual) recursion when the language lacks such support for expressing it explicitly (see https://en.wikipedia.org/wiki/Let_expression).

Now, this is not a problem in Nix, since its `let` expression is a `let-rec`, so the primary purpose it to allow extensibility of attribute sets (essentially [late binding](https://en.wikipedia.org/wiki/Late_binding)). In my understanding, this is also the motivation for this section, since `pkgs` heavily relay on fixed-point for overlays and other extension mechanisms.

Reply:
I didn't want to imply anything about `seq`. But all you said about let-rec is correct. I've boiled it down to minimize jargon.